lang/java/avro/src/main/java/org/apache/avro/generic/GenericDatumWriter.java

```diff
@@ -204,6 +205,10 @@ protected void writeField(Object datum, Field f, Encoder out, Object state) thro
     Object value = data.getField(datum, f.name(), f.pos(), state);
     try {
       write(f.schema(), value, out);
+    } catch (final UnresolvedUnionException uue) { // recreate it with the right field info
+      final UnresolvedUnionException unresolvedUnionException = new UnresolvedUnionException(f.schema(), f, value);
+      unresolvedUnionException.addSuppressed(uue);
```

Comment:
Hello!  Just curious -- why a suppressed exception here?

Reply:
To keep track of it without adding a wrapping level meaningless - at least was the idea